### PR TITLE
MULE-19388: ValueProviderModels from sdk-api usages must provide a SinceMuleVersionModelProperty

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

File: src/main/java/org/mule/runtime/api/meta/model/parameter/ValueProviderModel.java [hunk 1]
```diff
@@ -166,6 +215,24 @@ public int hashCode() {
...
+    checkArgument(propertyType != null, "Cannot get model properties of a null type");
+    return ofNullable((T) modelProperties.get(propertyType));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Set<ModelProperty> getModelProperties() {
+    return unmodifiableSet(new LinkedHashSet<>(modelProperties.values()));
```
Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/meta/model/parameter/ValueProviderModel.java:236]: don't copy on each invocation. Create the copy and the unmodifiable wrapper at the constructor itself

File: src/main/java/org/mule/runtime/api/meta/model/parameter/ValueProviderModel.java [hunk 2]
```diff
@@ -67,6 +80,41 @@ public ValueProviderModel(List<ActingParameterModel> parameters, boolean require
...
+    checkArgument(modelProperties != null, "'modelProperties' can't be null");
+    this.isOpen = isOpen;
+    this.parameters = unmodifiableList(parameters);
+    this.requiresConfiguration = requiresConfiguration;
+    this.requiresConnection = requiresConnection;
+    this.partOrder = partOrder;
+    this.providerName = providerName;
+    this.providerId = providerId;
+    this.modelProperties =
+        Arrays.stream(modelProperties).collect(toMap(modelProperty -> modelProperty.getClass(), modelProperty -> modelProperty));
```
Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/meta/model/parameter/ValueProviderModel.java:115]: static import

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/runtime/api/meta/model/parameter/ValueProviderModel.java [hunk 3]
```diff
@@ -166,6 +217,27 @@ public int hashCode() {
...
+    checkArgument(propertyType != null, "Cannot get model properties of a null type");
+    return ofNullable((T) modelProperties.get(propertyType));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Set<ModelProperty> getModelProperties() {
+    if (modelPropertiesValues == null) {
```
Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/meta/model/parameter/ValueProviderModel.java:238]: why the null check? this condition should be impossible

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/runtime/api/meta/model/parameter/ValueProviderModel.java (hunk 3 above)
Comment (ndinu) [reply src/main/java/org/mule/runtime/api/meta/model/parameter/ValueProviderModel.java:238]: If this is deserialized, wouldn't the modelPropertiesValues be null?

Comment (marianogonzalez) [reply src/main/java/org/mule/runtime/api/meta/model/parameter/ValueProviderModel.java:238]: that depends on the type adapter impl, but get your point

**Review by marianogonzalez — DISMISSED** [review]

**Review by SebaElizalde — APPROVED** [review]

**Pull request merged**